tests/test_s3_bucket_operations.py

```diff
@@ -0,0 +1,189 @@
+import logging
+import os
+from datetime import datetime, timedelta, timezone
+
+import pytest
+
+from noobaa_sa.exceptions import BucketAlreadyExists, BucketNotEmpty, NoSuchBucket
+
+log = logging.getLogger(__name__)
+
+
+class TestS3BucketOperations:
+    """
+    Test S3 bucket operations using NSFS
+    """
+
+    def test_bucket_creation_deletion_and_head(self, c_scope_s3client):
+        """
+        Test bucket creation and deletion via S3:
+        1. Create a bucket via S3
+        2. Verify the bucket was created via S3 HeadBucket
+        3. Delete the bucket via S3
+        4. Verify the bucket was deleted via S3 HeadBucket
+
+        """
+        # 1. Create a bucket via S3
+        bucket_name = c_scope_s3client.create_bucket()
+        # 2. Verify the bucket was created via S3 HeadBucket
+        assert c_scope_s3client.head_bucket(bucket_name), "Bucket was not created"
+        # 3. Delete the bucket via S3
+        c_scope_s3client.delete_bucket(bucket_name)
+        # 4. Verify the bucket was deleted via S3 HeadBucket
+        assert (
+            c_scope_s3client.head_bucket(bucket_name) == False
+        ), "Bucket was not deleted"
+
+    def test_list_buckets(self, c_scope_s3client):
+        """
+        Test listing buckets before creation and after deletion via S3
+
+        """
+        buckets, listed_buckets = [], []
+        AMOUNT = 5
+        try:
+            for _ in range(AMOUNT):
+                buckets.append(c_scope_s3client.create_bucket())
+
+            listed_buckets = c_scope_s3client.list_buckets()
+
+            # listed_buckets might contain buckets from before the test
+            assert all(
+                bucket in listed_buckets for bucket in buckets
+            ), "Created bucket was not listed!"
```

Comment:
Please use set operation here

Reply:
Done in https://github.com/red-hat-storage/noobaa-sa-ci/pull/16/commits/7f271a08a8fe91de47b2d8e10e4d553a0ed051d4